pkg/pool-utils/contracts/test/MockVault.sol

```diff
@@ -222,4 +222,9 @@ contract MockVault is IPoolSwapStructs {
     function setAuthorizer(IAuthorizer newAuthorizer) external {
         _authorizer = newAuthorizer;
     }
+
+    // Support VaultReentrancyLib
+    function manageUserBalance(IVault.UserBalanceOp[] memory ops) external payable {
+        // solhint-disable-previous-line no-empty-blocks
+    }
```

Comment:
Shouldn't we also have the reentrancy guard here? Tests that use the new lib will fail with this mock if not

Reply:
(also I don't think we need this change in this PR at all?)

Reply:
It's only here because it will be needed in multiple downstream PRs, which I wanted to keep independent.